src/OrchardCore/OrchardCore/Environment/Shell/Builders/ShellContainerFactory.cs

```diff
@@ -139,6 +139,21 @@ public IServiceProvider CreateContainer(ShellSettings settings, ShellBlueprint b
             return shellServiceProvider;
         }
 
+        private IServiceProvider GetProviderFromFactory(IServiceCollection collection)
+        {
+            var provider = collection.BuildServiceProvider(true);
+            var service = provider.GetService<IServiceFactoryAdapter>();
+            if (service == null)
+            {
+                return provider;
+            }
+
+            using (provider)
```

Comment:
Do we really need `using` here?!!

Reply:
I suppose so, since we are bootstrapping a factory generated provider and do not use it after that.